packages/rhi-webgl/src/WebGLRenderer.ts

```diff
@@ -188,53 +188,41 @@ export class WebGLRenderer implements IHardwareRenderer {
     this._gl.colorMask(r, g, b, a);
   }
 
-  clearRenderTarget(engine: Engine, clearMode: ClearMode, clearParam: Vector4) {
+  clearRenderTarget(
+    engine: Engine,
+    clearFlags: CameraClearFlags.Depth | CameraClearFlags.DepthColor,
+    clearColor: Color
+  ) {
     const gl = this._gl;
     const {
       blendState: { targetBlendState },
       depthState,
       stencilState
     } = engine._lastRenderState;
 
-    let clearFlag;
-    switch (clearMode) {
-      case ClearMode.SOLID_COLOR:
-        gl.clearColor(clearParam.x, clearParam.y, clearParam.z, clearParam.w);
-        clearFlag = gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT;
-        break;
-      case ClearMode.DEPTH_ONLY:
-        clearFlag = gl.DEPTH_BUFFER_BIT;
-        break;
-      case ClearMode.COLOR_ONLY:
-        gl.clearColor(clearParam.x, clearParam.y, clearParam.z, clearParam.w);
-        clearFlag = gl.COLOR_BUFFER_BIT;
-        break;
-      case ClearMode.STENCIL_ONLY:
-        gl.clear(gl.STENCIL_BUFFER_BIT);
-        clearFlag = gl.STENCIL_BUFFER_BIT;
-        break;
-      case ClearMode.ALL_CLEAR:
-        gl.clearColor(clearParam.x, clearParam.y, clearParam.z, clearParam.w);
-        clearFlag = gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT | gl.STENCIL_BUFFER_BIT;
-        break;
-    }
-
-    if (clearFlag & gl.COLOR_BUFFER_BIT && targetBlendState.colorWriteMask !== ColorWriteMask.All) {
-      gl.colorMask(true, true, true, true);
-      targetBlendState.colorWriteMask = ColorWriteMask.All;
+    let clearFlag = gl.DEPTH_BUFFER_BIT | gl.STENCIL_BUFFER_BIT;
+    if (clearFlags === CameraClearFlags.DepthColor) {
+      clearFlag = clearFlag | gl.COLOR_BUFFER_BIT;
+      if (clearColor.toGamma) {
```

Comment:
`if (clearColor.toGamma)` no useful, clearColor type is always `Color`

Reply:
+1